src/main/java/org/generationcp/middleware/service/impl/analysis/MeansImportRequest.java

```diff
@@ -0,0 +1,59 @@
+package org.generationcp.middleware.service.impl.analysis;
+
+import java.util.List;
+import java.util.Map;
+
+public class MeansImportRequest {
+
+	private Integer studyId;
+	private List<MeansData> data;
+
+	public Integer getStudyId() {
+		return this.studyId;
+	}
+
+	public void setStudyId(final Integer studyId) {
+		this.studyId = studyId;
+	}
+
+	public List<MeansData> getData() {
+		return this.data;
+	}
+
+	public void setData(final List<MeansData> data) {
+		this.data = data;
+	}
+
+	public static class MeansData {
+
+		private Integer environmentId;
```

Comment:
I think using `environmentNumber` (ie. trial instance 1, 2, 3..) will make more sense here than the environmentId (ie. nd_geolocation_id) because I think most if not all apps will not have straightforward access to the environment IDs - which are internal. 

In BV, the trial environment factor is usually the trial instance number of location name hence the result usually uses either of the two as trial environment identifier. I suspect the same for StABrAPP (please check)